src/program/latex/mod.rs

```diff
@@ -0,0 +1,1530 @@
+                    if let Some(open) = open {
+                        if let Some(close) = close {
+                            if ctrl < targ {
+                                // a vector with a head from the ctrl to the targ
+                                vector = 1 * (close - open);
+                            } else {
+                                // a vector with a head from the targ to the ctrl
+                                vector = -1 * (close - open);
+                            }
+                        }
+                    }
+                    // set wire at column as the control qubit of target qubit
+                    // computed as the distance from the control qubit
+                    self.circuit
+                        .get_mut(&ctrl)
+                        .and_then(|wire| wire.ctrl.insert(c, vector));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Takes a new or existing wire and adds or updates it using the name
+    /// (String) as the key. If a wire exists with the same name, then the
+    /// contents of the new wire are added to it by updating the next column
+    /// using the Quantikz command associated with its attributes (e.g. gate,
+    /// do_nothing, etc).
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes HashMap<String, Box<Circuit>>
+    /// `wire` - the wire to be pushed or updated to in circuits
+    fn push_wire(&mut self, wire: Wire) -> Result<(), LatexGenError> {
+        let qubit = wire.name;
+
+        // find wire in circuit collection
+        match self.circuit.get_mut(&wire.name) {
+            Some(wire_in_circuit) => {
+                // get the new gate from the wire and insert into existing wire
+                if let Some(gate) = wire.gates.get(&self.column) {
+                    // add gates to wire in circuit
+                    wire_in_circuit.gates.insert(self.column, gate.to_string());
+                }
+
+                // add modifiers to gate in circuit
+                if let Some(modifier) = wire.modifiers.get(&self.column) {
+                    wire_in_circuit
+                        .modifiers
+                        .insert(self.column, modifier.to_vec());
+                }
+
+                // add modifiers to gate in circuit
+                if let Some(parameters) = wire.parameters.get(&self.column) {
+                    wire_in_circuit
+                        .parameters
+                        .insert(self.column, parameters.to_vec());
+                }
+            }
+            _ => (),
+        }
+
+        // initalize relationships between multi qubit gates
+        if let Some(wire) = self.circuit.get(&qubit) {
+            // get the newly added gate if any at the column it was added
+            if let Some(gate) = wire.gates.get(&self.column) {
+                // tag relationships for multi qubit gates
+                if gate.starts_with('C') {
```

Comment:
This seems suspect. What happens if someone `DEFGATE`s a new gate with a name starting with `C`? Would be better to have an `is_controlled()` method or something to check this.